lumicks/pylake/kymotracker/detail/trace_line_2d.py

```diff
@@ -188,6 +188,29 @@ def in_rect(self, rect):
         coord_match = np.logical_and(coordinate < rect[1][1], coordinate >= rect[0][1])
         return np.any(np.logical_and(time_match, coord_match))
 
+    def sample_from_image(self, data, n_pixels, reduce=np.sum):
+        """Sample from image using coordinates from this KymoLine.
+
+        This function samples data from the image given in data based on the points in this KymoLine. It samples
+        from [time, position - n_pixels : position + pixels] and then applies the function sum.
+
+        Parameters
+        ----------
+        data : array_like
+            Image to sample from using this KymoLine.
+        n_pixels : int
+            Pixels in either direction to include in the sample
+        reduce : callable
+            Function evaluated on the sample. (Default: np.sum which produces sum of photon counts).
+        """
+        y_shape = data.shape[1]
```

Comment:
Pedantic: `y_size`? `shape` is usually a tuple of dimension sizes, not the individual dimension.